apps/hashdotdev/src/_pages/blog/0014_structured-data-value.mdx

```diff
@@ -0,0 +1,129 @@
+
+**Structured data can be used for verification and validation, in addition to improving the accuracy of a simulation's initial state.** Structured data can be used to backtest simulation experiments, as well as to test the accuracy and correctness of individual behaviors themselves. In a world where behaviors and whole simulation models may be generated by foundation models, having a concrete picture of reality to compare to provides humans with an improved sense of a model's likely correctness.
+
+### More for humans than machines
+
+All this serves to highlight an interesting possibility. If to date structured data has existed for the benefit of _machines_, is it possible that in the future its primary use will in fact be to us, _humans_, in verifying and validating the outputs of generative AI models: simulation, language, multimodal, or other?
+
+Inherently tied to the idea of structured data is the idea of **typing** information. Data may be of a certain 'type' if it follows a set of rules established by that type.
+
+The [Block Protocol type system](https://blockprotocol.org/docs/types) outlines three different kinds of types which can help us think about structured data: entity types, property types, and data types.
+
+1.  **Entities** (semantic 'things' in a database) have entity types.
+
+- **Entity types** describe the _properties_ and type(s) of any entity, including both special-cased _link entities_ and ordinary, plain old _standard entities_.
+- A **link entity** corresponds to the relationship, or 'thing' that connects other entities. For example, a standard entity `Person` may be connected by a link entity called `Spouse` to another standard entity `Person`.
+- A **standard entity** is absolutely anything that isn't a link entity. You can use it to refer to physical things, concepts, ideas, events, or anything else you like.
+- If you're familiar with graph theory, you can think of standard entities as vertices, and link entities as edges.
+
+2.  **Properties** contain data associated with some particular aspect of an entity.
+
+- **Property types** describe the possible values that can be associated with a property, by way of reference to _data types_ and other property types.
+
+3.  **Data types** describe a space of possible valid values (e.g. `string`, `number` or `boolean`). They allow for validation logic to be shared amongst different kinds of property types which represent semantically different things.
+
+- Support for user-definition of "non-primitive" data types (e.g. numbers within a certain range, or strings matching a certain pattern) which are not already found in the Block Protocol's [graph module specification](https://blockprotocol.org/docs/spec/graph) is planned ([draft RFC](https://github.com/blockprotocol/blockprotocol/pull/355)).
```

Comment:
I would say 'custom data types', or 'data types with custom constraints', or something like that... these are still primitive data types, but with validation constraints.
```suggestion
- We are also planning support for user-defined custom data types (e.g. numbers within a certain range, or strings matching a certain pattern) which are not already found in the Block Protocol's [graph module specification](https://blockprotocol.org/docs/spec/graph). Read the [draft RFC](https://github.com/blockprotocol/blockprotocol/pull/355) for more.
```